django/db/models/fields/mixins.py

```diff
@@ -1,31 +1,37 @@
 from django.core import checks
+from django.utils.functional import cached_property
 
 NOT_PROVIDED = object()
 
 
 class FieldCacheMixin:
-    """Provide an API for working with the model's fields value cache."""
+    """
+    Provide an API for working with the model's fields value cache.
 
-    def get_cache_name(self):
+    Subclasses must set self.cache_name to a unique entry for the cache -
+    typically the field’s name.
+    """
+
+    @cached_property
+    def cache_name(self):
         raise NotImplementedError
```

Comment:
Should we have a deprecation shim here that has `cache_name` call `get_cache_name` if not overriden?


```python
def get_cache_name(self):
    raise NotImplementedError

@cached_property
def cache_name(self):
    cache_name = self.get_cache_name()
    warnings.warn("cache_name must be overriden instead of get_cache_name", ...)
    return cache_name
```

Reply:
Done!